src/index.js

```diff
@@ -39,26 +39,35 @@ export default class SecoKeyval {
   }
 
   async set (key: string, val: any) {
-    if (!this.hasOpened) throw new Error('Must open first.')
-    this._data[key] = val
+    this.memset(key, val)
     await this.flush()
   }
 
-  get (key: string) {
+  async delete (key: string) {
+    const hadKey = this._data.hasOwnProperty(key)
+    this.memdelete(key)
+    if (hadKey) await this.flush()
+  }
+
+  memset (key: string, val: any) {
     if (!this.hasOpened) throw new Error('Must open first.')
-    return this._data[key]
+    this._data[key] = val
   }
 
-  async delete (key: string) {
+  memdelete (key: string) {
     if (!this.hasOpened) throw new Error('Must open first.')
 
     // Only need to delete and write if the key actually exists in the first place
     if (this._data.hasOwnProperty(key)) {
       delete this._data[key]
```

Comment:
maybe return `true` here to avoid the same check in set()

Reply:
I wondered about this myself, but when I actually benchmarked it, that check is so fast that removing it only made a difference of 2ms (70 -> 68 ms) on the benchmark in the test file (100,000 delete ops). I figured that kind of tiny perf improvement is probably not worth the additional lines of code

Reply:
ah i actually meant in terms of code, e.g.:
```js
set(key, value) {
  if (this.memset(key, value)) {
    await this.flush()
  }
}
```

but it's not important

Reply:
^^ oops, i meant for delete()

Reply:
Yeah same thing there - perf improvement of maybe a millisecond or two, which is probably not worth it when you compare it to the savings we're getting by using sync methods compared to `set` or `delete`. The benchmark in the test file would take upwards of 8 seconds using the async methods o.O